kpi/deployment_backends/kobocat_backend.py

```diff
@@ -159,9 +172,11 @@ def _kobocat_request(self, method, url, **kwargs):
                 detail=str(e), response=response)
 
         # Check for failure
-        if response.status_code != expected_status_code or (
-            'type' in json_response and json_response['type'] == 'alert-error'
-        ) or 'formid' not in json_response:
+        if (
+            response.status_code != expected_status_code
+            or json_response.get('type') == 'alert-error'
+            or not expect_formid and 'formid' not in json_response
```

Comment:
`or (expect_formid  and 'formid' not in json_response)`?
```
In [1]: r_stat_code = ex_stat_code = 200                                                                                             

In [2]: json_response = {'oops': 123}                                                                                                

In [3]: expect_formid = True                                                                                                         

In [4]: ( 
   ...:     r_stat_code != ex_stat_code 
   ...:     or json_response.get('type') == 'alert-error' 
   ...:     or not expect_formid and 'formid' not in json_response 
   ...: )                                                                                                                            
Out[4]: False  # but we want to enter the conditional block because the `expect_formid` test failed
```

Reply:
🤪

Reply:
please make more explicit with parentheses :pray: 
```python
        # Check for failure
        if (
            response.status_code != expected_status_code
            or json_response.get('type') == 'alert-error'
            or (expect_formid and 'formid' not in json_response)
        ):
```